src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-multisite-wizard/rgw-multisite-wizard.component.ts

```diff
@@ -0,0 +1,258 @@
+                frontendConfig: daemonStats['rgw_metadata']['frontend_config#0']
+              }))
+            )
+          );
+          return forkJoin(daemonStatsObservables);
+        })
+      )
+      .subscribe((daemonStatsArray) => {
+        this.rgwEndpoints.value = daemonStatsArray.map((daemonStats) => {
+          const protocol = daemonStats.frontendConfig.includes('ssl_port') ? 'https' : 'http';
+          return `${protocol}://${daemonStats.hostname}:${daemonStats.port}`;
+        });
+        const options: SelectOption[] = this.rgwEndpoints.value.map(
+          (endpoint: string) => new SelectOption(false, endpoint, '')
+        );
+        this.rgwEndpoints.options = [...options];
+      });
+
+    this.multiClusterService.getCluster().subscribe((clusters) => {
+      this.clusterDetailsArray = Object.values(clusters['config'])
+        .flat()
+        .filter((cluster) => cluster['cluster_alias'] !== 'local-cluster');
+      this.isMultiClusterConfigured = this.clusterDetailsArray.length > 0;
+      if (!this.isMultiClusterConfigured) {
+        this.stepTitles = ['Create Realm & Zonegroup', 'Create Zone', 'Export Multi-site token'];
+      } else {
+        this.selectedCluster = this.clusterDetailsArray[0]['name'];
+      }
+    });
+  }
+
+  createForm() {
+    this.multisiteSetupForm = new CdFormGroup({
+      realmName: new UntypedFormControl('default_realm', {
+        validators: [Validators.required]
+      }),
+      zonegroupName: new UntypedFormControl('default_zonegroup', {
+        validators: [Validators.required]
+      }),
+      zonegroup_endpoints: new UntypedFormControl(null, [Validators.required]),
+      zoneName: new UntypedFormControl('default_zone', {
+        validators: [Validators.required]
+      }),
+      zone_endpoints: new UntypedFormControl(null, {
+        validators: [Validators.required]
+      }),
+      username: new UntypedFormControl('default_system_user', {
+        validators: [Validators.required]
+      }),
+      cluster: new UntypedFormControl(null, {
+        validators: [Validators.required]
+      })
+    });
+
+    if (!this.isMultiClusterConfigured) {
+      this.exportTokenForm = new CdFormGroup({});
+    }
+  }
+
+  showSubmitButtonLabel() {
+    if (this.isMultiClusterConfigured) {
+      return !this.wizardStepsService.isLastStep()
+        ? this.actionLabels.NEXT
+        : $localize`Configure Multi-site`;
+    } else {
+      return !this.wizardStepsService.isLastStep() ? this.actionLabels.NEXT : $localize`Close`;
+    }
+  }
+
+  showCancelButtonLabel() {
+    return !this.wizardStepsService.isFirstStep()
+      ? this.actionLabels.BACK
+      : this.actionLabels.CANCEL;
+  }
+
+  onNextStep() {
+    if (!this.wizardStepsService.isLastStep()) {
+      this.wizardStepsService.getCurrentStep().subscribe((step: WizardStepModel) => {
+        this.currentStep = step;
+      });
+      if (this.currentStep.stepIndex === 2 && !this.isMultiClusterConfigured) {
```

Comment:
As this is a hardcoded value, if the steps are changed in the future this will fail. 
Maybe you could set the steps first depending on if `isMultiClusterConfigured` and then you only would nead to check if this is theLastStep to submit the form

Reply:
This is specific to multi-site automation wizard. This is not about checking if its theLastStep, because both the cases have equal number of steps i.e 3. This is because we need to perform the multi-site setup at the second step if its not MultiClusterConfigured and we need to do it at third step if its configured. Trying the wizard with both the cases can help it understand better.